### PR TITLE
Feat/instagram

### PR DESCRIPTION
**Pull request by stefanprokopdev**

### CODE REVIEW

File: src/lib/authist.ts [hunk 1]
```diff
@@ -2,15 +2,18 @@ import { NextFunction, Request, Response } from 'express';
...
 export const createAuthenticator = (options: AuthistOptions): Authenticator => ({
     expressBearer: expressBearer(options),
     signInWithEmailAndPassword: signInWithEmailAndPassword(options),
+    signInWithInstagram: signInWithInstagram(options),
 });
 
 export interface Authenticator {
     expressBearer: (request: Request, response: Response, next: NextFunction) => Promise<any>;
     signInWithEmailAndPassword: (email: string, password: string, req?: IncomingMessage) => Promise<UserCredentials>;
+    signInWithInstagram: (token: string, req?: IncomingMessage) => Promise<UserCredentials>;
```
Comment (smoliji) [inline src/lib/authist.ts]: What is this token? Please add a comment or readme docs. Not sure for instagram, but there can be several tokens (some acting like access tokens, some just only as a auth code you use in order to exchange it for the access token)

**Review by smoliji — CHANGES_REQUESTED** [review]

File: src/lib/authist.ts (hunk 1 above)
Comment (smoliji) [reply src/lib/authist.ts]: also have you considered using types from the provider directly? `ReturnType<typeof signInWithInstagram>` if its possible.. so you won't have to write it by hand :)

**Review by smoliji — CHANGES_REQUESTED** [review]

File: src/lib/authist.ts (hunk 1 above)
Comment (stefanprokopdev) [reply src/lib/authist.ts]: yes it's the facebook access token (any of the page / user / app / long lived tokens) I added links to the readme

I used the `ReturnType` it works fine! Thanks

Comment (stefanprokopdev) [reply src/lib/authist.ts]: I think I will merge some functionality with facebook provider later - it's the same api but the `fields` and `scopes` are different. There are also some differences between user token and page token

**Review by smoliji — APPROVED** [review]

**Pull request merged**